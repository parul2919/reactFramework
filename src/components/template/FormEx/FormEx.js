import React from "react";
import styled from "styled-components";
import styles from "./FormEx.style";
import MemoizedFooter from "../../organism/Footer/Footer";
import MemoizedHeader from "../../organism/Header/Header";
import AboutForm from "../../organism/AboutForm/AboutForm";

const FormEx = (props) => {
  const { className } = props;

  return (
    <div className={`${className} dummy-template`}>
      <MemoizedHeader />
      <AboutForm />
      <MemoizedFooter />
    </div>
  );
};

export default styled(FormEx)`
  ${styles}
`;
