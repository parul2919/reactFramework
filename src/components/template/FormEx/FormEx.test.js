import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import FormEx from "./FormEx";

describe("<Component /> spec", () => {
  it("renders the component", () => {
    const view = render(
      <Router>
        <FormEx />
      </Router>
    );
    expect(view).toMatchSnapshot();
  });
});
