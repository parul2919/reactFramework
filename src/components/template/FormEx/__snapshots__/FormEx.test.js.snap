// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ftvSup hsfLJE dummy-template"
      >
        <header
          class="header sc-eCYdqJ lkPLVQ"
        >
          <nav
            class="navbar  sc-hKMtZM jzgEMa"
          >
            <a
              class="navbar-brand "
              href="/"
            >
              <img
                alt="logo-img"
                src="test-file-stub"
                width="150"
              />
            </a>
            <div
              class="sc-gsnTZi gXHhKa collapse navbar-collapse menu-items"
            >
              <ul
                class="navbar-nav ml-auto main-menu "
              >
                <li
                  class="active dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/"
                    target="_self"
                  >
                    Tech Stacks
                  </a>
                </li>
                <li
                  class="inactive dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/component-library"
                    target="_self"
                  >
                    Component Library
                  </a>
                </li>
                <li
                  class="inactive dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/project-structure"
                    target="_self"
                  >
                    Project Structure
                  </a>
                </li>
                <li
                  class="inactive dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/utilities"
                    target="_self"
                  >
                    Utilities
                  </a>
                </li>
                <li
                  class="inactive dropdown-hide"
                >
                  <a
                    class="nav-link"
                    href="/testNav"
                    target="_self"
                  >
                    TestNav
                  </a>
                  <div
                    class="undefined collapse navbar-collapse sub-menu"
                  >
                    <ul
                      class="navbar-nav ml-auto sub-menu-1 "
                    >
                      <li
                        class="inactive dropdown-hide"
                      >
                        <a
                          class="nav-link"
                          href="/form"
                          target="_self"
                        >
                          FormEx
                        </a>
                      </li>
                      <li
                        class="inactive dropdown-hide"
                      >
                        <a
                          class="nav-link"
                          href="/2"
                          target="_self"
                        >
                          Subnav2
                        </a>
                        <div
                          class="undefined collapse navbar-collapse sub-menu"
                        >
                          <ul
                            class="navbar-nav ml-auto sub-menu-2 "
                          >
                            <li
                              class="inactive dropdown-hide"
                            >
                              <a
                                class="nav-link"
                                href="/3"
                                target="_self"
                              >
                                NestedNav
                              </a>
                            </li>
                          </ul>
                        </div>
                      </li>
                    </ul>
                  </div>
                </li>
              </ul>
            </div>
          </nav>
        </header>
        <div
          class="form-wrapper sc-iBkjds iFZfeR"
        >
          <form
            class="form-block"
          >
            <input
              class="sc-jSMfEi irIOZV"
              name="firstName"
              placeholder="first name"
            />
            <input
              class="sc-jSMfEi irIOZV ml-30"
              name="lastName"
              placeholder="last name"
            />
            <input
              class="sc-jSMfEi irIOZV"
              name="department"
              placeholder="department"
            />
            <select
              class="sc-gKXOVf dRMlYe ml-30"
              name="gender"
            >
              <option
                value="female"
              >
                female
              </option>
              <option
                value="male"
              >
                male
              </option>
              <option
                value="other"
              >
                other
              </option>
            </select>
            <input
              class="submit"
              type="submit"
            />
          </form>
        </div>
        <section
          class="footer sc-bczRLJ eRSPWO"
        >
          <nav
            style="max-width: 1440px; margin: auto;"
          >
            <a
              class="footer-logo"
              href="/"
            >
              <img
                alt="logo-img"
                src="test-file-stub"
                width="150"
              />
            </a>
          </nav>
          <div
            class="disclaimer-text text-center"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ips um has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ftvSup hsfLJE dummy-template"
    >
      <header
        class="header sc-eCYdqJ lkPLVQ"
      >
        <nav
          class="navbar  sc-hKMtZM jzgEMa"
        >
          <a
            class="navbar-brand "
            href="/"
          >
            <img
              alt="logo-img"
              src="test-file-stub"
              width="150"
            />
          </a>
          <div
            class="sc-gsnTZi gXHhKa collapse navbar-collapse menu-items"
          >
            <ul
              class="navbar-nav ml-auto main-menu "
            >
              <li
                class="active dropdown-hide"
              >
                <a
                  class="nav-link"
                  href="/"
                  target="_self"
                >
                  Tech Stacks
                </a>
              </li>
              <li
                class="inactive dropdown-hide"
              >
                <a
                  class="nav-link"
                  href="/component-library"
                  target="_self"
                >
                  Component Library
                </a>
              </li>
              <li
                class="inactive dropdown-hide"
              >
                <a
                  class="nav-link"
                  href="/project-structure"
                  target="_self"
                >
                  Project Structure
                </a>
              </li>
              <li
                class="inactive dropdown-hide"
              >
                <a
                  class="nav-link"
                  href="/utilities"
                  target="_self"
                >
                  Utilities
                </a>
              </li>
              <li
                class="inactive dropdown-hide"
              >
                <a
                  class="nav-link"
                  href="/testNav"
                  target="_self"
                >
                  TestNav
                </a>
                <div
                  class="undefined collapse navbar-collapse sub-menu"
                >
                  <ul
                    class="navbar-nav ml-auto sub-menu-1 "
                  >
                    <li
                      class="inactive dropdown-hide"
                    >
                      <a
                        class="nav-link"
                        href="/form"
                        target="_self"
                      >
                        FormEx
                      </a>
                    </li>
                    <li
                      class="inactive dropdown-hide"
                    >
                      <a
                        class="nav-link"
                        href="/2"
                        target="_self"
                      >
                        Subnav2
                      </a>
                      <div
                        class="undefined collapse navbar-collapse sub-menu"
                      >
                        <ul
                          class="navbar-nav ml-auto sub-menu-2 "
                        >
                          <li
                            class="inactive dropdown-hide"
                          >
                            <a
                              class="nav-link"
                              href="/3"
                              target="_self"
                            >
                              NestedNav
                            </a>
                          </li>
                        </ul>
                      </div>
                    </li>
                  </ul>
                </div>
              </li>
            </ul>
          </div>
        </nav>
      </header>
      <div
        class="form-wrapper sc-iBkjds iFZfeR"
      >
        <form
          class="form-block"
        >
          <input
            class="sc-jSMfEi irIOZV"
            name="firstName"
            placeholder="first name"
          />
          <input
            class="sc-jSMfEi irIOZV ml-30"
            name="lastName"
            placeholder="last name"
          />
          <input
            class="sc-jSMfEi irIOZV"
            name="department"
            placeholder="department"
          />
          <select
            class="sc-gKXOVf dRMlYe ml-30"
            name="gender"
          >
            <option
              value="female"
            >
              female
            </option>
            <option
              value="male"
            >
              male
            </option>
            <option
              value="other"
            >
              other
            </option>
          </select>
          <input
            class="submit"
            type="submit"
          />
        </form>
      </div>
      <section
        class="footer sc-bczRLJ eRSPWO"
      >
        <nav
          style="max-width: 1440px; margin: auto;"
        >
          <a
            class="footer-logo"
            href="/"
          >
            <img
              alt="logo-img"
              src="test-file-stub"
              width="150"
            />
          </a>
        </nav>
        <div
          class="disclaimer-text text-center"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ips um has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
